framework/platforms/core-platform/src/main/java/io/fairyproject/container/object/resolver/ContainerObjectResolverImpl.java

```diff
@@ -73,4 +74,24 @@ public class ContainerObjectResolverImpl implements ContainerObjectResolver {
 
         return future;
     }
+
+    @Override
+    public @NotNull CompletableFuture<Object> resolveInstance(@NotNull TypeDescriptor typeDescriptor) throws Exception {
+        ContainerObj object = this.binder.getBinding(typeDescriptor);
+        ConditionUtils.notNull(object, String.format("Couldn't find container object %s!", typeDescriptor));
+
+        CompletableFuture<Object> future;
+        switch (object.getScope()) {
+            case SINGLETON:
+                future = singletonObjectFactory.createInstance(object.getType());
```

Comment:
Both SingletonObjectFactory and PrototypeObjectFactory does not take TypeDescriptor into account, this may cause issue when there are multiple instances with the same class but different generic.